pkg/agent/openflow/network_policy.go

```diff
@@ -1979,17 +1998,17 @@ func (c *client) NetworkPolicyMetrics() map[uint32]*types.RuleMetric {
 	}
 	if c.enableMulticast {
 		// We need to collect NP statistics matching IGMP query messages and egress multicast traffic.
-		collectMetricsFromFlows(MulticastIngressMetricTable, parseMulticastMetricFlow)
-		collectMetricsFromFlows(MulticastEgressMetricTable, parseMulticastMetricFlow)
+		collectMetricsFromFlows(MulticastIngressMetricTable, []string{metricFlowIdentifier, "reg3"}, nil, parseMulticastMetricFlow)
+		collectMetricsFromFlows(MulticastEgressMetricTable, []string{metricFlowIdentifier, "reg3"}, nil, parseMulticastMetricFlow)
 	}
 	// We have two flows for each allow rule. One matches 'ct_state=+new'
 	// and counts the number of first packets, which is also the number
 	// of sessions (this is the reason why we have 2 flows). The other
 	// matches 'ct_state=-new' and is used to count all subsequent
 	// packets in the session. We need to merge metrics from these 2
 	// flows to get the correct number of total packets.
-	collectMetricsFromFlows(EgressMetricTable, parseMetricFlow)
-	collectMetricsFromFlows(IngressMetricTable, parseMetricFlow)
+	collectMetricsFromFlows(EgressMetricTable, []string{metricFlowIdentifier}, []string{"reg0", "ct_label"}, parseMetricFlow)
+	collectMetricsFromFlows(IngressMetricTable, []string{metricFlowIdentifier}, []string{"reg0", "ct_label"}, parseMetricFlow)
```

Comment:
The original way was not very proper and efficient to identify metric flows. I feel adding more common fields that are not specific to metric flows is still error-prone and not very readable.
Could you consider using cookie to fix the problem once for all? If we use a separate categary or some bits in cookie to indicate metric flows, we could get the exact metric flows by specifying cookie when querying the flows, no need to filter flows in antrea-agent.